### PR TITLE
Fixing bug QSVT wire order

### PR DESCRIPTION
**Pull request by KetpuntoG**
**Context:** 
The order of the wires in QSVT is not maintained as they are used as a set internally.

**issue**
Solves issue #5965

### CODE REVIEW

File: doc/releases/changelog-dev.md [hunk 1]
```diff
@@ -28,10 +28,15 @@
 
 <h3>Bug fixes 🐛</h3>
 
+* `qml.QSVT` was not storing the order of the wires correctly.
```
Comment (soranjh) [inline doc/releases/changelog-dev.md:31]:
```suggestion
* `qml.QSVT` is updated to store wire order correctly.
```

File: pennylane/templates/subroutines/qsvt.py [hunk 2]
```diff
@@ -285,10 +285,12 @@ def __init__(self, UA, projectors, id=None):
             "projectors": projectors,
         }
 
-        ua_wires = UA.wires.toset()
-        proj_wires = set.union(*(proj.wires.toset() for proj in projectors))
+        ua_wires = qml.wires.Wires(UA.wires)
```
Comment (soranjh) [inline pennylane/templates/subroutines/qsvt.py:288]: Could it be more compact? e.g., as

`total_wires = UA.wires + sum(proj.wires for proj in projectors)`

**Review by soranjh — CHANGES_REQUESTED** [review]

File: pennylane/templates/subroutines/qsvt.py (hunk 2 above)
Comment (KetpuntoG) [reply pennylane/templates/subroutines/qsvt.py:288]: For some reason, doing that adds the wires 0 even if I'm not using it. 🤔

Comment (KetpuntoG) [reply pennylane/templates/subroutines/qsvt.py]: solved

File: pennylane/templates/subroutines/qsvt.py [hunk 3]
```diff
@@ -285,10 +285,10 @@ def __init__(self, UA, projectors, id=None):
             "projectors": projectors,
         }
 
-        ua_wires = UA.wires.toset()
-        proj_wires = set.union(*(proj.wires.toset() for proj in projectors))
+        total_wires = qml.wires.Wires(UA.wires) + qml.wires.Wires.all_wires(
```
Comment (soranjh) [inline pennylane/templates/subroutines/qsvt.py:288]: You might import Wires to make the code more compact:
`total_wires = Wires(UA.wires) + Wires.all_wires([proj.wires for proj in projectors])`

File: doc/releases/changelog-dev.md [hunk 4]
```diff
@@ -34,12 +34,18 @@
...
+  [(#5959)](https://github.com/PennyLaneAI/pennylane/pull/5959)
+
 * `qml.AmplitudeEmbedding` has better support for features using low precision integer data types.
 [(#5969)](https://github.com/PennyLaneAI/pennylane/pull/5969)
 
+
 <h3>Contributors ✍️</h3>
 
 This release contains contributions from (in alphabetical order):
+
```
Comment (soranjh) [inline doc/releases/changelog-dev.md:47]: Why these blank lines are added?

**Review by soranjh — APPROVED** [review]

File: doc/releases/changelog-dev.md (hunk 4 above)
Comment (KetpuntoG) [inline doc/releases/changelog-dev.md:47]:
```suggestion
```

File: doc/releases/changelog-dev.md [hunk 5]
```diff
@@ -34,12 +34,17 @@
 
 <h3>Bug fixes 🐛</h3>
 
+* `qml.QSVT` is updated to store wire order correctly.
+  [(#5959)](https://github.com/PennyLaneAI/pennylane/pull/5959)
+
 * `qml.AmplitudeEmbedding` has better support for features using low precision integer data types.
 [(#5969)](https://github.com/PennyLaneAI/pennylane/pull/5969)
 
+
```
Comment (KetpuntoG) [inline doc/releases/changelog-dev.md:43]:
```suggestion
```

File: doc/releases/changelog-dev.md [hunk 6]
```diff
@@ -8,8 +8,8 @@
 
 * The `qml.PrepSelPrep` template is added. The template implements a block-encoding of a linear 
   combination of unitaries.
-  [(#5756)](https://github.com/PennyLaneAI/pennylane/pull/5756)
-  
+  [(#5756)](https://github.com/PennyLaneAI/pennylane/pull/5756), [(#5987)](https://github.com/PennyLaneAI/pennylane/pull/5987)
+
```
Comment (KetpuntoG) [inline doc/releases/changelog-dev.md:12]:
```suggestion
  [(#5756)](https://github.com/PennyLaneAI/pennylane/pull/5756)
```

File: doc/releases/changelog-dev.md [hunk 7]
```diff
@@ -9,7 +9,6 @@
 * The `qml.PrepSelPrep` template is added. The template implements a block-encoding of a linear 
   combination of unitaries.
   [(#5756)](https://github.com/PennyLaneAI/pennylane/pull/5756)
```
Comment (KetpuntoG) [inline doc/releases/changelog-dev.md:11]:
```suggestion
  [(#5756)](https://github.com/PennyLaneAI/pennylane/pull/5756)
  
```

File: doc/releases/changelog-dev.md (hunk 4 above)
Comment (KetpuntoG) [reply doc/releases/changelog-dev.md]: removed

**Review by Jaybsoni — APPROVED** [review]
Looks good to me! 

I don't really think this is a bug? Since when we decompose the operations will act on the wires in the correct fashion, and matrix_expand will compute the matrix for each operation before multiplying them according to required wire_order. So its mostly just a visual artifact, the wires look like they are in a certain order, but actually it doesn't matter.

**Comment by KetpuntoG** [comment]
> Looks good to me!
> 
> I don't really think this is a bug? Since when we decompose the operations will act on the wires in the correct fashion, and matrix_expand will compute the matrix for each operation before multiplying them according to required wire_order. So its mostly just a visual artifact, the wires look like they are in a certain order, but actually it doesn't matter.

I detected the issue because I was not getting the correct results. This solves the issue 😄

**Comment by KetpuntoG** [comment]
[sc-68059]

**Pull request merged**